source/tools-and-extensions/archiving/HDB++.rst

```diff
@@ -48,6 +56,195 @@ Elements
+.. _hdbpp-es: https://github.com/tango-controls-hdbpp/hdbpp-es
+.. _hdbpp-cm-es: https://github.com/tango-controls-hdbpp/hdbpp-cm-es
+.. _libhdbpp: https://github.com/tango-controls-hdbpp/libhdbpp
+.. _libhdbpp-mysql: https://github.com/tango-controls-hdbpp/libhdbpp-mysql
+.. _libhdbpp-mysql-legacy: https://github.com/tango-controls-hdbpp/libhdbpp-mysql-legacy
+.. _libhdbpp-cassandra: https://github.com/tango-controls-hdbpp/libhdbpp-cassandra
+.. _CassandraMonitor: https://github.com/tango-controls-hdbpp/CassandraMonitor
+.. _hdbpp-configurator: https://github.com/tango-controls-hdbpp/hdbpp-configurator
+.. _hdbpp-viewer: https://github.com/tango-controls-hdbpp/hdbpp-viewer
+.. _libhdbpp-extraction-java: https://github.com/tango-controls-hdbpp/libhdbpp-extraction-java
+.. _libhdbpp-extraction-cpp: https://github.com/tango-controls-hdbpp/libhdbpp-extraction-cpp
+.. _eGiga2m: https://github.com/luciozambon/eGiga2m
+
+* hdbpp-cm_ : the HDB++ Configuration Manager device server
+* hdbpp-es_: the HDB++ Event Subscriber device server
+* hdbpp-cm-es_: Device server able to export HDB++ Event Subscriber and Configuration Manager devices in the same device server. It is a multiclass Tango device server capable of both CongigurationManager and EventSubscriber devices. Can be used in place of hdbpp-cm and hdbpp-es devices.
+* libhdbpp_: the HDB++ insertion (abstract) library
+* libhdbpp-mysql_: the HDB++ insertion library for MySQL backend
+* libhdbpp-mysql-legacy_: the HDB++ insertion library for MySQL backend using the old Tango HDB database schema
+* libhdbpp-cassandra_: the HDB++ insertion library for Cassandra backend
+* CassandraMonitor_: A Java client/server to monitor cassandra nodes using jmx calls.
+* hdbpp-configurator_: the HDB++ Configuration GUI (in Java).
+* hdbpp-viewer_: the HDB++ Viewer GUI (in Java)
+* libhdbpp-extraction-java_: HDB++ Java extraction library
+* libhdbpp-extraction-cpp_: HDB++ C++ extraction library
+* eGiga2m_: Web graphic data viewer able to show HDB++ data
+
+.. note::
+    Please, find README file in each repository. It contains a lot of necessary information.
+
+
+
+Documentation for building and installing
+-----------------------------------------
+
+The documentation for building and installing many of the components is available on the
+different git repositories from https://github.com/tango-controls-hdbpp Github organization.
+
+
+Please, have a look at the README files from the different git repositories
+(For instance: https://github.com/tango-controls-hdbpp/hdbpp-cm-es).
+They explain how to install the HDB++ libraries as well as the device servers.
+
+The latest versions of *hdbpp-es* and *hdbpp-cm* are now loading dynamically (using dlopen) the HDB++ library used for MySQL/legacy HDB schema, MySQL HDB++ schema, Cassandra.
+
+
+.. note::
+    You can use "-v5" argument to see the error as "-v5" sets the logging level of the device to DEBUG. By doing this, DEBUG level logs will be printed on your terminal.
```

Comment:
-v5 can be used to see the debug messages coming from the Tango library itself too. This can be very verbose.
-v4 can be used to see the DEBUG messages coming from the Tango devices themselves.
In recent versions of libhdb++cassandra library, there are some configuration parameters to adjust libhdb++cassandra library log level and the cassandra driver log level.
These configuration parameters are parameters of LibConfiguration Tango class or device property.
These parameters are named differently depending on the version of the library used (logging_enabled or logging_level and cassandra_driver_log_level)